### PR TITLE
feat(curriculum): add regex lecture transcripts

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by Nirajn2311 — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733aad43b3ebff588a26fb5.md [hunk 1]
```diff
@@ -1,14 +1,140 @@
 ---
 id: 6733aad43b3ebff588a26fb5
 title: What Are Regular Expressions, and What Are Some Common Methods?
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733aad43b3ebff588a26fb5.md:3]: finished editing for this one. need to check the video before uploading to youtube. We can merge this in for now. But there will need to be another PR to add in the id.

File: curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5ba834ded4bb067e67c.md [hunk 2]
```diff
@@ -1,14 +1,156 @@
 ---
 id: 6733c5ba834ded4bb067e67c
 title: What Are Some Common Regular Expression Modifiers Used for Searching?
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5ba834ded4bb067e67c.md:3]: finished editing for this one. need to check the video before uploading to youtube. We can merge this in for now. But there will need to be another PR to add in the id.

**Review by jdwilkin4 — APPROVED** [review]
